AnkiDroid/src/main/java/com/ichi2/libanki/sched/AbstractSched.java

```diff
@@ -234,4 +239,25 @@ public String toString() {
     public interface CountMethod {
         int operation(long did, int lim);
     }
+
+    protected static void leech(Card card, Activity activity) {
+        if (activity != null) {
+            Resources res = activity.getResources();
+            final String leechMessage;
+            if (card.getQueue() < 0) {
+                leechMessage = res.getString(R.string.leech_suspend_notification);
+            } else {
+                leechMessage = res.getString(R.string.leech_notification);
+            }
+            activity.runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    Toast.makeText(activity, leechMessage, Toast.LENGTH_SHORT).show();
+                }
+            });
+
+        } else {
+            Timber.e("LeechHook :: could not show leech toast as activity was null");
```

Comment:
Ideally leave in a separate commit as it's a functionality change - has been an issue for some time.

```suggestion
            Timber.w("LeechHook :: could not show leech toast as activity was null");
```

Reply:
I'm okay with adding this change if it please you. I've no idea what commit message to write. I do not know the reason of this change. Can you write the message please ?

Reply:
LOG or NF prefix:

`NF: Leech - Reduced Log Level`